src/transport_ofi.c

```diff
@@ -315,9 +358,46 @@ int bind_enable_cntr_ep_resources(shmem_transport_ctx_t *ctx)
 {
     int ret = 0;
 
-    /* Attach the shared context */
-    ret = fi_ep_bind(ctx->cntr_ep, &ctx->stx->fid, 0);
-    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind STX to CNTR endpoint failed");
+    /* STX contexts are shared by SHMEM contexts that are private to the same
+     * thread (i.e. have SHMEM_CTX_PRIVATE option set and same thread ID).  */
+    if (ctx->options & SHMEM_CTX_PRIVATE) {
+        shmem_transport_ofi_stx_kvs_t *e = malloc(sizeof(shmem_transport_ofi_stx_kvs_t));
+        e->tid = ctx->tid;
+        e->stxs = malloc(sizeof(shmem_transport_ofi_stx_t));
+
+        shmem_transport_ofi_stx_kvs_t *f;
+        HASH_FIND_INT(shmem_transport_ofi_stx_kvs, &ctx->tid, f);
```

Comment:
I think it would make more sense the check the hash table for an entry for the given TID before constructing a new entry.  If there is an existing entry, you can just increment its ref_cnt rather than replacing it with a new entry.

Reply:
👍